### ISSUE TITLE
docker_service "Failed to import docker-py"

### ISSUE THREAD

**Issue by d3v1an7** [ansible/ansible-modules-core#5435]
##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

docker_service

##### ANSIBLE VERSION

```
ansible 2.3.0
  config file =
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

No custom config
##### OS / ENVIRONMENT

OS X 10.11.6
##### SUMMARY

Ansible unable to locate docker-py, even though it's installed (see version output below).

**Versions:**

```
$ pip show docker-py

---
Metadata-Version: 2.0
Name: docker-py
Version: 1.10.5
Summary: Python client for Docker.
Home-page: https://github.com/docker/docker-py/
Author: Joffrey F
Author-email: joffrey@docker.com
Installer: pip
License: UNKNOWN
Location: /usr/local/lib/python2.7/site-packages
Requires: six, backports.ssl-match-hostname, ipaddress, websocket-client, docker-pycreds, requests
Classifiers:
  Development Status :: 4 - Beta
  Environment :: Other Environment
  Intended Audience :: Developers
  Operating System :: OS Independent
  Programming Language :: Python
  Programming Language :: Python :: 2
  Programming Language :: Python :: 2.6
  Programming Language :: Python :: 2.7
  Programming Language :: Python :: 3
  Programming Language :: Python :: 3.3
  Programming Language :: Python :: 3.4
  Programming Language :: Python :: 3.5
  Topic :: Utilities
  License :: OSI Approved :: Apache Software License
```

```
$ docker version
Client:
 Version:      1.12.1
 API version:  1.24
 Go version:   go1.7.1
 Git commit:   6f9534c
 Built:        Thu Sep  8 10:31:18 2016
 OS/Arch:      darwin/amd64

Server:
 Version:      1.12.1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   23cf638
 Built:        Thu Aug 18 17:52:38 2016
 OS/Arch:      linux/amd64
```

```
$ docker-compose --version
docker-compose version 1.8.0, build f3628c7
```
##### STEPS TO REPRODUCE

See example playbook:

```

---
- name: test docker service
  docker_service:
    project_src: "{{ lookup('env', 'HOME') }}/docker-compose-yaml-file-in-this-dir"
```
##### EXPECTED RESULTS

docker_service to read/execute docker-compose.yaml file successfully
##### ACTUAL RESULTS

ansible exits after task fails

```
fatal: [localhost]: FAILED! => {
    "changed": false,
    "failed": true,
    "invocation": {
        "module_args": {
            "api_version": null,
            "build": false,
            "cacert_path": null,
            "cert_path": null,
            "debug": false,
            "definition": null,
            "dependencies": true,
            "docker_host": null,
            "files": null,
            "filter_logger": false,
            "hostname_check": false,
            "key_path": null,
            "nocache": false,
            "project_name": null,
            "project_src": "/Users/hahaha/docker-compose-yaml-file-in-this-dir",
            "pull": false,
            "recreate": "smart",
            "remove_images": null,
            "remove_orphans": false,
            "remove_volumes": false,
            "restarted": false,
            "scale": null,
            "services": null,
            "ssl_version": null,
            "state": "present",
            "stopped": false,
            "timeout": 10,
            "tls": null,
            "tls_hostname": null,
            "tls_verify": null
        },
        "module_name": "docker_service"
    },
    "msg": "Failed to import docker-py - No module named requests.exceptions. Try `pip install docker-py`"
}
```

**Comment by d3v1an7** [comment]
FYI, the error when running the same playbook with Ansible  2.1.2.0 is the following:

```
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/var/folders/93/pzh7rm456fb56vv96ljx60h00000gq/T/ansible_2wRT6c/ansible_module_docker_service.py", line 408, in <module>
    import yaml
ImportError: No module named yaml

fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_name": "docker_service"}, "module_stderr": "Traceback (most recent call last):\n  File \"/var/folders/93/pzh7rm456fb56vv96ljx60h00000gq/T/ansible_2wRT6c/ansible_module_docker_service.py\", line 408, in <module>\n    import yaml\nImportError: No module named yaml\n", "module_stdout": "", "msg": "MODULE FAILURE"}
```

And yep, `pyyaml` is definitely installed :)

```
$ pip show pyyaml
---
Metadata-Version: 1.1
Name: PyYAML
Version: 3.12
Summary: YAML parser and emitter for Python
Home-page: http://pyyaml.org/wiki/PyYAML
Author: Kirill Simonov
Author-email: xi@resolvent.net
License: MIT
Location: /usr/local/lib/python2.7/site-packages
Requires:
Classifiers:
  Development Status :: 5 - Production/Stable
  Intended Audience :: Developers
  License :: OSI Approved :: MIT License
  Operating System :: OS Independent
  Programming Language :: Python
  Programming Language :: Python :: 2
  Programming Language :: Python :: 2.7
  Programming Language :: Python :: 3
  Programming Language :: Python :: 3.4
  Programming Language :: Python :: 3.5
  Topic :: Software Development :: Libraries :: Python Modules
  Topic :: Text Processing :: Markup
```

**Comment by ansibot** [comment]
@ansible, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by prata** [comment]
I've run into this same issue:

"msg": "Failed to import docker-py - No module named requests.exceptions. Try `pip install docker-py`"

[prata@kusanagi playbooks]$ pip show docker-py; docker version; ansible --version; docker-compose --version

Name: docker-py
Version: 1.10.6
Summary: Python client for Docker.
Home-page: https://github.com/docker/docker-py/
Author: Joffrey F
Author-email: joffrey@docker.com
License: UNKNOWN
Location: /usr/local/lib/python2.7/site-packages
Requires: requests, websocket-client, backports.ssl-match-hostname, ipaddress, six, docker-pycreds
Client:
 Version:      1.12.1
 API version:  1.24
 Go version:   go1.7.1
 Git commit:   6f9534c
 Built:        Thu Sep  8 10:31:18 2016
 OS/Arch:      darwin/amd64

Server:
 Version:      1.12.1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   23cf638
 Built:        Thu Aug 18 17:52:38 2016
 OS/Arch:      linux/amd64

docker-compose version 1.8.1, build 878cff1

ansible 2.1.2.0
  config file = /Users/prata/.ansible.cfg
  configured module search path = ['/usr/share/ansible']

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jctanner** [comment]
I just ran into a similar issue:

```
"msg": "Failed to import docker-py - cannot import name InvalidHeader. Try `pip install docker-py`"
```

Please clean out all copies of the requests library, reinstall it and then re-test ansible.

needs_info

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by d3v1an7** [comment]
Not migrating ticket, as now running with `shell` module instead. Sorry future ticket reader!

**Comment by derekmahar** [comment]
@d3v1an7 So do you now use the `shell` module to invoke docker and no longer use any of the `docker_*` Ansible modules?

**Comment by IoTPlay** [comment]
Same problem. `And pip docker-py` is installed. I am on a Mac.
> FAILED! => {"changed": false, "msg": "Failed to import docker-py - No module named websocket. Try `pip install docker-py`"}

**Comment by dagwieers** [comment]
This repository is no longer used or tracked. If you still have an issue, please open a new issue at: https://github.com/ansible/ansible/issues/new

**Comment by IoTPlay** [comment]
I did create a new issue, then you guys closed that as well.

"This is not a problem with ansible, but instead a potential problem with upstream docker-py/docker":

docker/docker-py#1353

where to from here?
